# Quick start for SGS genome browser
## Installation of SGS
```shell
# download packages
wget "https://xxxxx/xxx.tar.gz"
gunzip xxx.tar.gz
./flutter_smart_genome
```



## Deploy new SGS server
SGS browser is mainly composed of two parts:SGS server and SGS client. After downloading and installing the SGS client, we also need to deploy SGS server for data visuliation. Here，we have developed a one-click deployment function that simplifies the installation steps, allowing users to quickly deploy their own native SGS services.

Note that only Linux or MacOS supports this function.


### First step:clicking the one-click deploy button
We need to open the SGS browser client and click the one-click deploy button, then we can choose this computer or remote computer to install SGS service:
![one-click](/pictures/3_1.png)


### Second step:Filling relevant information.
We need to provide server ip address, port, username, password information for SGS server installation. Then, entering the deployment interface and filling the root password again. 


#### Deploy on this computer
Deploy on this computer only need to offer root password.
![one-click](/pictures/3_2.png)

#### Deploy on remote computer
+ IP: the server IP address, we can view it by "ifconfig -a" command.
+ username: the name of user
+ password: the password of the server
![one-click](/pictures/3_3.png)


### Finish SGS deploy
After installation, we can see the web server, R server and local server.
![one-click](/pictures/3_4.png)



## Adding species
To add a new species, you need to click the "add species" botton. Then, You should fill in the corresponding species name and description information, and choose a fasta file in the data dir.
![Adding species](/pictures/4_6.png)




## Adding tracks
SGS support three types of tracks, namely Genome、Transcript(single cell transcipt) and ATAC(single cell atac). To add tracks, you first need to choose track types, then load files. Users can add single file or multi files at the same time.
![Adding tracks](/pictures/4_7.png)



## Adding data from Rstudio server
### Installation sgsload Package
In order to realize the joint exploration of data analysis and visualization, we developed a sgsload R package to directly import the single-cell analysis objects generated by the Rstudio server analysis environment.

```R
# download package
# install from github
devtools::install_github("bio-xtt/sgsload") 

# install from source
install.packages("/home/bio-xtt/Desktop/sgsload_0.1.0.tar.gz", type=source)
```


### Getting specices id
Note that before importing single-cell analysis objects directly into the SGS single-cell browser, you need to add a species fasta file and view the id of the species.
```R
# load the package
library(sgsload)

# get the speices id information, the default userid is "user001".
get_species_inform(user_id)
```


### Loadding Seurat object into SGS cellbrowser
Loadding a scRNA object created by Seurat into SGS cellbrowser.
```R
# load the example datasets: pbmc_small
data("pbmc_small")

# load the example marker genes: marker.tsv
marker_file <- system.file("extdata/scRNA", "markers.tsv", package = "sgsload")

# use the export function
result_json <- ExportSC(
  object = pbmc_small,
  species_id = "25542c58734641fe98ba9f6b51d17a20",
  track_name = "pbmc_small",
  track_type = "scRNA",
  select_group = c("groups", "RNA_snn_res.0.8", "letter.idents", "cluster"),
  assays = "RNA",
  matrix.slot = list("RNA" = "data"),
  assay.type = list("RNA" = "gene"),
  reductions = c("tsne", "umap"),
  marker.files = list("RNA" = marker_file))
```


### Loadding Signac object into SGS cellbrowser
Loadding a scATAC object created by Signac into SGS cellbrowser.
```R
# load the package
library(sgsload)

# get the speices id information
get_species_inform(user_id)

# load the example datasets: pbmc_small
data("atac")

# load the example marker files created by different assay: marker_gene.tsv、marker_motif.tsv、marker_peak.tsv
marker_genes <- system.file("extdata/scATAC", "marker_genes.tsv", package = "sgsload")
marker_motifs <- system.file("extdata/scATAC", "marker_motif.tsv", package = "sgsload")
marker_peaks <- system.file("extdata/scATAC", "marker_peaks.tsv", package = "sgsload")

# create a marker file list and name the list with the correlated assay name
markers <- list(marker_genes, marker_motifs, marker_peaks)
names(markers) <- c("RNA", "chromvar", "peaks")

# if you changed the fragments file path, you should update the fragments file path sorted in the signac object
new.path <- system.file("extdata/scATAC", "fragments.tsv.gz", package = "sgsload")
fragments <- CreateFragmentObject(
  path = new.path,
  cells = colnames(atac),
  validate.fragments = TRUE )
Fragments(atac) <- NULL
Fragments(atac) <- fragments

# use the export function to loadding the scATAC object into SGS cellbrowser
result_json2 <- ExportSC(
  object = atac,
  species_id = "25542c58734641fe98ba9f6b51d17a20",
  track_name = "atac_demo",
  track_type = "scATAC",
  select_group = c("seurat_clusters"),
  assays = c("RNA", "chromvar", "peaks"),
  matrix.slot = list("RNA" = "data", "chromvar" = "data", "peaks" = "data"),
  assay.type = list("RNA" = "gene", "chromvar" = "motif", "peaks" = "peak"),
  reductions = "umap",
  marker.files = markers)
```